test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/testapp/integration/auth/customizer/CustomTokenConverterTest.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.testapp.integration.auth.customizer
+
+import org.junit.jupiter.api.Test
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Import
+import uk.gov.justice.digital.hmpps.testapp.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.testapp.resource.CustomAuthAwareAuthenticationTokenConverter
+import uk.gov.justice.hmpps.kotlin.auth.dsl.ResourceServerConfigurationCustomizer
+
+@Import(CustomTokenConverterTest.CustomizerConfiguration::class)
+@SpringBootTest(
+  webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+  // allow overriding of the ResourceServerConfigurationCustomizer bean definition
+  properties = ["spring.main.allow-bean-definition-overriding=true"],
+)
+class CustomTokenConverterTest : IntegrationTestBase() {
+
+  private class CaseloadFinder {
+    fun find() = "MDI"
+  }
+
+  @TestConfiguration
+  class CustomizerConfiguration {
+    @Bean
+    fun resourceServerCustomizer() = ResourceServerConfigurationCustomizer {
+      // Add a custom token converter that gets caseload from "somewhere", e.g. could be a bean calling another service
+      oauth2 { tokenConverter = CustomAuthAwareAuthenticationTokenConverter { CaseloadFinder().find() } }
+    }
+  }
+
+  @Test
+  fun `should return OK for an unauthorized endpoint`() {
+    webTestClient.get()
+      .uri("/info")
+      .exchange()
+      .expectStatus()
+      .isOk
+  }
+
+  @Test
+  fun `should return unauthorized for a protected endpoint`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden without a role`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .headers(setAuthorisation())
+      .exchange()
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `should return forbidden with wrong role`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+      .exchange()
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `should exercise the custom token converter`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .headers(setAuthorisation(roles = listOf("ROLE_CUSTOM_CONVERTER")))
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBody()
+      .jsonPath("activeCaseload").isEqualTo("MDI")
+  }
```

Comment:
This test uses the token converter to grab the active caseload and returns it - so proves that the custom token converter is being used and works as expected.